app/over_react_redux/todo_client/lib/src/components/app.dart

```diff
@@ -1,101 +1,135 @@
 import 'package:over_react/over_react.dart';
 import 'package:over_react/over_react_redux.dart';
 import 'package:over_react/src/component/test_fixtures/redraw_counter_component_mixin.dart';
+import 'package:react_material_ui/react_material_ui.dart';
 
 import 'package:todo_client/src/actions.dart';
+import 'package:todo_client/src/components/app_bar/app_bar_theme_menu.dart';
 import 'package:todo_client/src/store.dart';
 import 'package:todo_client/src/models/todo.dart';
 import 'package:todo_client/src/models/user.dart';
 import 'package:todo_client/src/components/app_bar/app_bar.dart';
 import 'package:todo_client/src/components/create_input.dart';
-import 'package:react_material_ui/react_material_ui.dart';
 import 'package:todo_client/src/components/todo_list.dart';
 import 'package:todo_client/src/components/user_list.dart';
 
 part 'app.over_react.g.dart';
 
 UiFactory<TodoAppProps> TodoApp = connect<AppState, TodoAppProps>(
-    mapDispatchToProps: (dispatch) {
-      return (TodoApp()
-        ..createTodo = (description) {
-          dispatch(AddTodoAction(Todo(description: description)));
-        }
-        ..createUser = (name) {
-          dispatch(AddUserAction(User(name: name)));
-        }
-      );
-    },
-    forwardRef: true,
+  mapDispatchToProps: (dispatch) {
+    return (TodoApp()
+      ..createTodo = (description) {
+        dispatch(AddTodoAction(Todo(description: description)));
+      }
+      ..createUser = (name) {
+        dispatch(AddUserAction(User(name: name)));
+      });
+  },
+  mapStateToProps: (state) {
+    return TodoApp()..theme = state.theme;
+  },
+  forwardRef: true,
 )(_$TodoApp); // ignore: undefined_identifier
 
 mixin TodoAppPropsMixin on UiProps {
   Function(String description) createTodo;
 
   Function(String name) createUser;
+
+  String theme;
```

Comment:
Just for my learning, would you also be able to type this prop as a `MuiTheme`? I can see why it being a `String` here would be easier since it's also used as a state, but if the theme didn't change based on user input, would it work to have this as a `MuiTheme`? Or are there other reasons to use a string theme prop and create the actual `MuiTheme` object in the component?

Reply:
Yep, this prop could be a `MuiTheme` - like you said, I decided it was easier to make it a `String` since the default app state is initialized as JSON and it was just easier to avoid converting the `String` to the class to pass as a prop.

Reply:
Awesome! Thanks!